### PR TITLE
New TF GLUE example

### PR DESCRIPTION
**Pull request by Rocketknight1**
This is the PR for the new-style TF GLUE example. I'd like to run a few more tests, especially on the weirder datasets like MNLI and STSB, before I merge it, but it's almost ready!

### CODE REVIEW

File: examples/tensorflow/text-classification/run_glue.py [hunk 1]
```diff
@@ -0,0 +1,554 @@
...
+    # region Load model config and tokenizer
+    #
+    # In distributed training, the .from_pretrained methods guarantee that only one local process can concurrently
+    # download model & vocab.
+    config = AutoConfig.from_pretrained(
+        model_args.config_name if model_args.config_name else model_args.model_name_or_path,
+        num_labels=num_labels,
+        finetuning_task=data_args.task_name,
+        cache_dir=model_args.cache_dir,
+        revision=model_args.model_revision,
```
Comment (sgugger) [inline examples/tensorflow/text-classification/run_glue.py:319]: We should set the label2id and id2label here as well (so that when finetuned the model has the proper labels on the inference API).

(This is also needed on the PyTorch side, will make a PR soon-ish)

**Review by sgugger — APPROVED** [review]
Looking great! Thanks for doing this!

**Review by LysandreJik — APPROVED** [review]
Very cool, thanks for working on it @Rocketknight1!

File: examples/tensorflow/text-classification/run_glue.py [hunk 2]
```diff
@@ -368,6 +352,10 @@ def main():
                 f"model labels: {list(sorted(label_name_to_id.keys()))}, dataset labels: {list(sorted(label_list))}."
                 "\nIgnoring the model labels as a result.",
             )
+            label_to_id = {label: i for i, label in enumerate(label_list)}
+    if label_to_id is not None:
+        config.label2id = label_to_id
+    config.id2label = {id: label for label, id in config.label2id.items()}
```
Comment (sgugger) [inline examples/tensorflow/text-classification/run_glue.py:357]:
```suggestion
        config.id2label = {id: label for label, id in config.label2id.items()}
```
No point doing that otherwise I think.

Comment (Rocketknight1) [reply examples/tensorflow/text-classification/run_glue.py:357]: Fastest reviewing ever!
